rs/rosetta-api/icrc1/ledger/sm-tests/src/lib.rs

```diff
@@ -2536,29 +2539,39 @@ pub fn icrc1_test_approval_upgrade<T>(
             }
         }
         assert_eq!(expected_allowances, allowances);
+
+        for account in &all_accounts {
+            assert_eq!(balance_of(&env, canister_id, *account), balances[account]);
+        }
     };
 
-    // Test if the old serialized approvals are correctly deserialized
-    test_upgrade(ledger_wasm_current.clone());
-    // Test if new approvals serialization also works
-    test_upgrade(ledger_wasm_current);
+    // Test if the old serialized approvals and balances are correctly deserialized
+    test_upgrade(ledger_wasm_current.clone(), balances.clone());
+    // Test the new wasm serialization
+    test_upgrade(ledger_wasm_current.clone(), balances.clone());
+    // Test serializing to the memory manager
+    test_upgrade(ledger_wasm_upgradetomemorymanager, balances.clone());
+    // Test deserializing from memory manager
+    test_upgrade(ledger_wasm_current, balances.clone());
 
     // Add some more approvals
     for a1 in &accounts {
         for a2 in &additional_accounts {
             let mut approve_args = default_approve_args(*a2, APPROVE_AMOUNT);
             approve_args.from_subaccount = a1.subaccount;
             send_approval(&env, canister_id, a1.owner, &approve_args).expect("approval failed");
+            balances.insert(a1, balances[a1].clone() - approve_args.fee.unwrap());
```

Comment:
the fee can also be none, should it not be the default ledger fee?

Reply:
Since we specify the fee in the `approve_args`, the test will fail anyway if the ledger fee is None?